### ISSUE TITLE
Build error `Error in bail mode: Error: callback(): The callback was already called` and how to mitigate

### ISSUE THREAD

**Issue by hlovdal** [SimonAlling/userscripter#94]
### TL;DR

If you get error `Error in bail mode: Error: callback(): The callback was already called` when running `npm run build` then you need to set the `NODE_OPTIONS` environment variable to include `--openssl-legacy-provider`.

### Details

I created a test project by running `node .../src/github/userscripter/bin/cli.js init` followed by running `npm install` and `npm run build`, and the build command failed with

```text
ℹ ｢atl｣: Using tsconfig.json from .../userscript/test1/tsconfig.json
Error in bail mode: Error: callback(): The callback was already called.
    at context.callback (.../userscript/test1/node_modules/loader-runner/lib/LoaderRunner.js:106:10) Error: callback(): The callback was already called.
    at context.callback (.../userscript/test1/node_modules/loader-runner/lib/LoaderRunner.js:106:10)
```

After making some debug modifications to `node_modules/loader-runner/lib/LoaderRunner.js` I noticed the following in the debug output

```text
    error: ModuleParseError: Module parse failed: error:0308010C:digital envelope routines::unsupported
    File was processed with these loaders:
     * ./node_modules/awesome-typescript-loader/dist/entry.js
     * ./node_modules/restrict-imports-loader/dist/index.js
    You may need an additional loader to handle the result of these loaders.
    Error: error:0308010C:digital envelope routines::unsupported
        at new Hash (node:internal/crypto/hash:71:19)
        at Object.createHash (node:crypto:130:10)
        at module.exports (.../userscript/test1/node_modules/webpack/lib/util/createHash.js:135:53)
        at NormalModule._initBuildHash (.../userscript/test1/node_modules/webpack/lib/NormalModule.js:417:16)
        at handleParseResult (.../userscript/test1/node_modules/webpack/lib/NormalModule.js:477:10)
        at .../userscript/test1/node_modules/webpack/lib/NormalModule.js:500:6
        at .../userscript/test1/node_modules/webpack/lib/NormalModule.js:358:12
        ...
```

which matched the part about "digital envelope routines::unsupported" from [this comment](https://github.com/Zaid-Ajaj/fable-getting-started/issues/32#issuecomment-1304933171), and when I ran 

```bash
env NODE_OPTIONS=--openssl-legacy-provider npm run build
```

then the build succeeded.

```bash
$ npm --version
9.6.0
$ node --version
v16.18.1
```

**Comment by SimonAlling** [comment]
Awesome! That's been a major pain point for me as well lately. I don't remember exactly what the context was, but the next time I face this error, I'll keep this issue in mind!

**Comment by SimonAlling** [comment]
Yeah, this happens with Node 18 and 20 btw, but not with Node 16.

**Comment by SimonAlling** [comment]
> Yeah, this happens with Node 18 and 20 btw, but not with Node 16.

Yeah, maybe that reply of mine wasn't very helpful, given that it _did_ happen with Node 16 for @hlovdal – specifically:

```console
$ npm --version
9.6.0
$ node --version
v16.18.1
```

But it doesn't happen for me:

```console
$ npm --version
8.19.2
$ node --version
v16.18.1
```

Also doesn't happen for me with v16.20.2, but it does happen with v18.0.0. I'm on my ThinkPad X1 Carbon 5th Gen with Ubuntu 24.04.

My repro for reference:

```
git clone https://github.com/SimonAlling/example-userscript
cd example-userscript/
git checkout d1d30ea54062a620179784159f9a922d30ecc696
npm ci
npm run build
```
